event/oracle/upgrade_oracle_event.go

```diff
@@ -0,0 +1,173 @@
+package oracle
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/edgelesssys/ego/enclave"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/crypto"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/panacea"
+	"github.com/medibloc/panacea-doracle/sgx"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type UpgradeOracleEvent struct {
+	reactor event.Reactor
+}
+
+var _ event.Event = (*UpgradeOracleEvent)(nil)
+
+func NewUpgradeOracleEvent(s event.Reactor) UpgradeOracleEvent {
+	return UpgradeOracleEvent{s}
+}
+
+func (e UpgradeOracleEvent) GetEventType() string {
+	return "message"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeKey() string {
+	return "action"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeValue() string {
+	return "'OracleUpgrade'"
+}
+
+func (e UpgradeOracleEvent) EventHandler(event ctypes.ResultEvent) error {
+	uniqueID := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyUniqueID][0]
+	votingTargetAddress := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyOracleAddress][0]
+
+	msgVoteOracleRegistration, err := e.verifyAndGetMsgVoteOracleRegistration(uniqueID, votingTargetAddress)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("oracle upgrade voting info. uniqueID(%s), voterAddress(%s), votingTargetAddress(%s), voteOption(%s)",
+		msgVoteOracleRegistration.OracleRegistrationVote.UniqueId,
+		msgVoteOracleRegistration.OracleRegistrationVote.VoterAddress,
+		msgVoteOracleRegistration.OracleRegistrationVote.VotingTargetAddress,
+		msgVoteOracleRegistration.OracleRegistrationVote.VoteOption,
+	)
+
+	txBuilder := panacea.NewTxBuilder(*e.reactor.QueryClient())
+	txBytes, err := txBuilder.GenerateTxBytes(e.reactor.OracleAcc().GetPrivKey(), e.reactor.Config(), msgVoteOracleRegistration)
+	if err != nil {
+		return err
+	}
+
+	if err := e.broadcastTx(txBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e UpgradeOracleEvent) verifyAndGetMsgVoteOracleRegistration(uniqueID, votingTargetAddress string) (*oracletypes.MsgVoteOracleRegistration, error) {
+	queryClient := e.reactor.QueryClient()
+	voterAddress := e.reactor.OracleAcc().GetAddress()
+	oraclePrivKeyBz := e.reactor.OraclePrivKey().Serialize()
+
+	oracleRegistration, err := queryClient.GetOracleRegistration(votingTargetAddress, uniqueID)
+	if err != nil {
+		log.Infof("failed to get oracleRegistration. uniqueID(%s), address(%s). %v", uniqueID, votingTargetAddress, err)
+		return makeMsgVoteOracleRegistrationVoteTypeNo(uniqueID, voterAddress, votingTargetAddress, oraclePrivKeyBz)
+	}
+
+	voteOption, err := e.verifyAndGetVoteOption(oracleRegistration)
+	if err != nil {
+		log.Infof("vote No due to error while verify: %v", err)
```

Comment:
maybe missed the below
```go
return makeMsgVoteOracleRegistrationVoteTypeNo(uniqueID, voterAddress, votingTargetAddress, oraclePrivKeyBz)
```

Reply:
I right.
As we discussed before, if error occurs voteOption might be `NO`.
No issue here. Thx!

Reply:
But it looks good to voting `No` this line.
I will fixed it.